Core/src/org/sleuthkit/autopsy/modules/photoreccarver/PhotoRecCarverFileIngestModule.java

```diff
@@ -435,17 +441,32 @@ synchronized Path createModuleOutputDirectoryForCase() throws IngestModule.Inges
      *
      * @throws IngestModuleException
      */
-    public static File locateExecutable(String executableToFindName) throws IngestModule.IngestModuleException {
-        // Must be running under a Windows operating system.
-        if (!PlatformUtil.isWindowsOS()) {
-            throw new IngestModule.IngestModuleException(Bundle.unsupportedOS_message());
+    public static File locateExecutable() throws IngestModule.IngestModuleException, IOException {
+        File exeFile = null;
+        Path execName = null;
+        String photorec_linux_directory = "/usr/bin";
+        if (PlatformUtil.isWindowsOS()) {
+            execName = Paths.get(PHOTOREC_DIRECTORY, PHOTOREC_EXECUTABLE);
+            exeFile = InstalledFileLocator.getDefault().locate(execName.toString(), PhotoRecCarverFileIngestModule.class.getPackage().getName(), false);
+        } else {
+            File usrBin = new File("/usr/bin/photorec");
+            File usrLocalBin = new File("/usr/local/bin/photorec");
+            if (usrBin.canExecute() && usrBin.exists() && !usrBin.isDirectory()) {
+                photorec_linux_directory = "/usr/bin";
+            }else if(usrLocalBin.canExecute() && usrLocalBin.exists() && !usrLocalBin.isDirectory()){
+                photorec_linux_directory = "/usr/local/bin";
+            }else{
+                exeFile = null;
```

Comment:
This should throw IngestModule.IngestModuleException with message that photorec exe can not be found.